component/src/main/java/org/wso2/extension/siddhi/execution/env/ResourceBatchWindowProcessor.java

```diff
@@ -0,0 +1,383 @@
+                    windowLength = ResourceIdentifierStreamProcessor.getResourceCount(resourceName);
+                } else {
+                    throw new SiddhiAppValidationException(
+                            "Resource Batch window's 'resource.group.id' parameter should be String, but found "
+                                    + attributeExpressionExecutors[1].getReturnType());
+                }
+            } else {
+                throw new SiddhiAppValidationException("Resource Batch window should have constant "
+                        + "for 'resource.group.id' parameter but found a dynamic attribute " +
+                        attributeExpressionExecutors[1].getClass().getCanonicalName());
+            }
+            this.groupKeyExpressionExecutor = attributeExpressionExecutors[1];
+            if (attributeExpressionExecutors.length == 3) {
+                this.groupKeyExpressionExecutor = attributeExpressionExecutors[1];
+                if ((attributeExpressionExecutors[2] instanceof ConstantExpressionExecutor)) {
+                    if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.INT) {
+                        timeInMilliSeconds = Integer.parseInt(String.valueOf(((ConstantExpressionExecutor)
+                                attributeExpressionExecutors[2]).getValue()));
+                    } else if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.LONG) {
+                        timeInMilliSeconds = Long.parseLong(String.valueOf(((ConstantExpressionExecutor)
+                                attributeExpressionExecutors[2]).getValue()));
+                    } else {
+                        throw new SiddhiAppValidationException("ResourceBatch window's 3rd parameter " +
+                                "'time.in.milliseconds' should be either be a constant (of type int or long), " +
+                                "but found " + attributeExpressionExecutors[2].getReturnType());
+                    }
+                } else {
+                    throw new SiddhiAppValidationException("ResourceBatch window's 3rd parameter " +
+                            "'time.in.milliseconds' should either be a constant (of type int or long)");
+                }
+            }
+        } else {
+            throw new SiddhiAppValidationException(
+                    "Resource batch window should only have two or three parameters, but found "
+                            + attributeExpressionExecutors.length + " input attributes");
+        }
+    }
+
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor,
+                           StreamEventCloner streamEventCloner) {
+        synchronized (this) {
+            long currentTime = siddhiAppContext.getTimestampGenerator().currentTime();
+            ComplexEventChunk<StreamEvent> outputStreamEventChunk = new ComplexEventChunk<StreamEvent>(true);
```

Comment:
add all the eventsToBeExpired to the  outputStreamEventChunk and clear that

Reply:
Fixes added: d6242876caf0f65c7fd023c25c7c5e50e9fd7b3c